packages/amplify-graphql-api-construct/src/internal/data-source-config.ts

```diff
@@ -93,3 +93,24 @@ export const getDataSourceStrategiesProvider = (definition: IAmplifyGraphqlDefin
 
   return provider;
 };
+
+/**
+ * Validates the user input for the dataSourceStrategy. This is a no-op for DynamoDB strategies for now.
+ * @param strategy user provided model data source strategy
+ * @returns validates and throws an error if the strategy is invalid
+ */
+export const validateDataSourceStrategy = (strategy: ConstructModelDataSourceStrategy) => {
+  if (!isSqlStrategy(strategy)) {
+    return;
+  }
+
+  const dbConnectionConfig = strategy.dbConnectionConfig;
+  const invalidSSMPaths = Object.values(dbConnectionConfig).filter((value) => typeof value === 'string' && !isValidSSMPath(value));
+  if (invalidSSMPaths.length > 0) {
+    throw new Error(`Invalid SSM paths in dbConnectionConfig: ${invalidSSMPaths.join(', ')}. SSM paths must start with '/'.`);
```

Comment:
I think including the data source strategy name in the error message would also be useful to customers.

Reply:
+1, especially since `combine` explicitly supports multiple strategy definitions

Reply:
Updated to include the strategy name as well.